src/pj_internal.c

```diff
@@ -134,36 +136,193 @@ size_t pj_strlcpy(char *dst, const char *src, size_t siz) {
+char *pj_shrink (char *c) {
+/****************************************************************************************
+    Clash repeated whitespace, remove '+' and ';', make ',' and '=' greedy, eating
+    their surrounding whitespace.
+****************************************************************************************/
+    size_t i, j, n, ws;
+
+    if (0==c)
+       return 0;
+
+    pj_chomp (c);
+    n = strlen (c);
+
+    /* First clash repeated whitespace (including +/;) */
+    for (i = j = ws = 0;  j < n;  j++) {
+
+        /* Blank out prefix '+', only if preceeded by whitespace (i.e. keep it in 1.23e+08 */
+        if ((i > 0) && ('+'==c[j]) && isspace (c[i]))
+            c[j] = ' ';
+
+        if (isspace (c[j]) || ';'==c[j] || '\\'==c[j]) {
+            if (0==ws && (i > 0))
+                c[i++] = ' ';
+            ws = 1;
+            continue;
+        }
+        else
+            ws = 0;
+        c[i++] = c[j];
+    }
+    c[i] = 0;
+    n = strlen(c);
+
+    /* Then make ',' and '=' greedy */
+    for (i = j = 0; j < n; j++) {
+        if (i==0) {
+            c[i++] = c[j];
+            continue;
+        }
+
+        /* Skip space before '='/',' */
+        if ('='==c[j] || ','==c[j]) {
+            if (c[i - 1]==' ')
+               c[i - 1] = c[j];
+            else
+                c[i++] = c[j];
+            continue;
+        }
+
+        if (' '==c[j] && ('='==c[i - 1] || ','==c[i - 1]) )
+            continue;
+
+        c[i++] = c[j];
+    }
+    c[i] = 0;
+    return c;
 }
 
 
 
-/* stuff below is *not* considered API, and will be moved to an "internal plumbing toolset" */
+
+/***************************************************************************************/
+size_t pj_trim_args (char *args) {
+/****************************************************************************************
+    Trim an argument string and count its number of elements.
+****************************************************************************************/
+    size_t i, m, n;
+    pj_shrink (args);
+    n = strlen (args);
+    if (n==0)
+        return 0;
+    for (i = m = 0;  i < n;  i++) {
+        if (' '==args[i]) {
+            args[i] = 0;
+            m++;
+        }
+    }
+    return m + 1;
+}
+
+
+/***************************************************************************************/
+char **pj_trimmed_args_to_argv (size_t argc, char *args) {
+/****************************************************************************************
+    Create an argv-style array from elements placed in args.
+
+    args is a trimmed string as returned by pj_trim_args(), and argv is the number of
+    trimmed strings found (i.e. the return value of pj_trim_args()).
+****************************************************************************************/
+    size_t i, j;
+    char **argv;
+    /* turn the massaged input into an array of strings */
```

Comment:
By following your previous advice, I think this has become more clear from the (improved) context. To avoid any remaining possibilities of confusion, the comment now reads *turn the input string into an array of strings*